### DISCUSSION TITLE
Debugging Model in razor theme views

### DISCUSSION THREAD

**Discussion by nogginbox** [OrchardCMS/OrchardCore#8342] [Q&A] [answered]
To debug my Orchard views I've been adding break points in visual studio and then adding watches to the Model to see what's in it. As Model is a shape much of this is dynamic and it's is not always obvious what properties are available on something like Model.Content.

What is thes best way to view the data and available properties in Model?

**Comment by deanmarcussen** [comment, marked as answer]
The Shape does have a DebuggerView, but until you know Shapes, and the variety of properties that are in them it does not always have much meaning.

Personally I don't like using the debugger on Razor views a lot, I find it takes too long to get it running, and doesn't add enough when it does.

The `Orchard.ConsoleLog(Model.Content as object)` helper which dumps shapes and other objects as json to the console is the tool I tend to make the most use off.

Also available for liquid as a filter `{{ Model.Content | console_log}}`

It is best to call it after calling `DisplayAsync(Model.Content)` for shapes so that all the display events fire, and populate alternates etc.

Also will render `@Orchard.ConsoleLog(Model.ContentItem as object)` content items, and other objects.
